src/lexer/token_mapping.cpp

```diff
@@ -303,6 +304,25 @@ static std::map<std::string, DefinitionType> extern_definitions = {
     {"nrn_random_play", DefinitionType::EXT_5}};
 
 
+/**
+ * Checks if \c token is one of the functions coming from NEURON/CoreNEURON and needs
+ * passing NrnThread* as first argument (typical name of variable \c nt)
+ *
+ * @param token Name of function
+ * @return True or false depending if the function needs NrnThread* argument
+ */
+bool needs_neuron_thread_first_arg(const std::string& token) {
+    auto extern_def = extern_definitions.find(token);
+    if (extern_def != details::extern_definitions.end() &&
+        (extern_def->second == details::DefinitionType::EXT_4 ||
+         extern_def->second == details::DefinitionType::EXT_DOUBLE_4)) {
+        return true;
+    } else {
+        return false;
+    }
```

Comment:
This writing is probably simpler:

```
return extern_def != details::extern_definitions.end() &&
        (extern_def->second == details::DefinitionType::EXT_4 ||
         extern_def->second == details::DefinitionType::EXT_DOUBLE_4);
```

The if is unnecessary

Reply:
you can even write this:

```
bool needs_neuron_thread_first_arg(const std::string& token) inline {
return extern_definitions.find(token) != details::extern_definitions.end() &&
        (extern_def->second == details::DefinitionType::EXT_4 ||
         extern_def->second == details::DefinitionType::EXT_DOUBLE_4);
}
```

and put it in the hpp as inliner

Reply:
Thanks for the suggestions 👍

Reply:
I decided in the end to avoid inlining the function to avoid exposing more details to the header file but still good suggestion!